File: consumer/src/main/java/com/codeforprofit/redis/consumer/config/RedisConfiguration.java
package com.codeforprofit.redis.consumer.config;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.redis.connection.RedisConnectionFactory;
import org.springframework.data.redis.connection.stream.Consumer;
import org.springframework.data.redis.connection.stream.MapRecord;
import org.springframework.data.redis.connection.stream.ReadOffset;
import org.springframework.data.redis.connection.stream.StreamOffset;
import org.springframework.data.redis.stream.StreamMessageListenerContainer;

@Slf4j
@Configuration
public class RedisConfiguration {

    @Bean
    public  StreamMessageListenerContainer<String, MapRecord<String, String, String>> messageListener(RedisConnectionFactory factory)  {
        final StreamMessageListenerContainer<String, MapRecord<String, String, String>> container =
            StreamMessageListenerContainer.create(factory);
        container.receive(Consumer.from("group2", "name2"), StreamOffset.create("stream1", ReadOffset.lastConsumed()), message -> log.info(message.toString()));
        container.start();

        return container;
    }
}

File: producer/src/main/kotlin/com/codeforprofit/redis/producer/config/RedisConfiguration.kt
package com.codeforprofit.redis.producer.config

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.connection.ReactiveRedisConnectionFactory
import org.springframework.data.redis.connection.RedisConnectionFactory
import org.springframework.data.redis.connection.stream.*
import org.springframework.data.redis.core.ReactiveRedisOperations
import org.springframework.data.redis.core.ReactiveRedisTemplate
import org.springframework.data.redis.serializer.RedisSerializationContext
import org.springframework.data.redis.stream.StreamMessageListenerContainer

@Configuration
@EnableConfigurationProperties(RedisProperties::class)
class RedisConfiguration(redisProperties: RedisProperties) {

    @Bean
    fun reactiveRedisOperations(factory: ReactiveRedisConnectionFactory): ReactiveRedisOperations<String, String> {
        return ReactiveRedisTemplate<String, String>(factory, RedisSerializationContext.string())
    }

    @Bean
    fun messageListener(factory: RedisConnectionFactory): StreamMessageListenerContainer<String, MapRecord<String, String, String>> {
        return StreamMessageListenerContainer.create(factory)
            .apply {
                receive(Consumer.from("group1", "name1"), StreamOffset.create("stream1", ReadOffset.lastConsumed()), { message -> println(message) })

                start()
            }
    }
}
